test_pytest_html.py

```diff
@@ -287,12 +287,74 @@ def pytest_runtest_makereport(item, call):
                     report.extra = [extras.image('{0}')]
         """.format(content))
         testdir.makepyfile('def test_pass(): pass')
-        result, html = run(testdir)
+        result, html = run(testdir, 'report.html', '--self-contained-html')
         assert result.ret == 0
         assert '<a class="image" href="#" target="_blank">Image</a>' in html
         src = 'data:image/png;base64,{0}'.format(content)
         assert '<a href="#"><img src="{0}"/></a>'.format(src) in html
 
+    def test_extra_image_separeted(self, testdir):
+        content = b64encode(str(random.random())
+                            .encode('utf-8')).decode('ascii')
+        testdir.makeconftest("""
+            import pytest
+            @pytest.mark.hookwrapper
+            def pytest_runtest_makereport(item, call):
+                outcome = yield
+                report = outcome.get_result()
+                if report.when == 'call':
+                    from pytest_html import extras
+                    report.extra = [extras.image('{0}')]
+        """.format(content))
+        testdir.makepyfile('def test_pass(): pass')
+        result, html = run(testdir)
+        hash_key = ('test_extra_image_separeted.py::'
+                    'test_pass0-1').encode('utf-8')
```

Comment:
I'm having second thoughts on the `-1` after seeing it here. Functionally it's fine but it looks ugly to me. 😝 I suppose we could increment the reruns so they're never `0`. I'll let you decide on that. 😀